### PR TITLE
devel/llvm-base: new ports to add /usr/bin/cc etc

### PR DESCRIPTION
**Pull request by brooksdavis**
This ports depends on a another devel/llvm* port and adds wrappers
for cc, c++, cpp, ld, ld.lld, and objdump in /usr/bin so that program
not looking for a specific compiler can find them.  On CHERI
architectures the wrapper adds compiler flags so that the wrappers
target the expected ABI.

### CODE REVIEW

**Comment by brooksdavis** [comment]
I've verified that the wrapper works as expected on amd64 including using the `UNAME_p` environment variable to trick the script into asking like it's on morello.

**Comment by brooksdavis** [comment]
Reworked at @jrtc27's suggestion to just depend on devel/llvm

File: devel/llvm-base/files/wrapper.sh.in [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+realtool=${LOCALBASE}/bin/${llvmtool}
+
+# CheriBSD: assume that if we're on a CHERI architecture we want either
+# purecap or hybrid binaries.
+arch_cflags=
+arch_objdump_flags=
+ARCH=`uname -p`
+case $ARCH in
+aarch64)
+	arch_cflags="-target aarch64-unknown-freebsd -march=morello+a64c"
```
Comment (jrtc27) [inline devel/llvm-base/files/wrapper.sh.in:74]: Are we concerned this doesn't include the OS version? Also the triple should be picked up correctly as the default anyway? We might want to use a Morello LLVM that's new enough to support `-march=morello -mabi=purecap` and just use `-march=morello` everywhere (already worked fine for hybrid, a64c was the default before I made the mode inferred), that way `cc -mabi=aapcs` and `cc -mabi=purecap` both work, modulo emutls, without needing to override -march.

File: devel/llvm-base/files/wrapper.sh.in [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	arch_objdump_flags="--triple=aarch64 --mattr=+morello"
```
Comment (jrtc27) [inline devel/llvm-base/files/wrapper.sh.in:75]: Same as cflags, do we need the triple?

File: devel/llvm-base/pkg-descr [hunk 3]
```diff
@@ -0,0 +1,8 @@
+The LLVM Project is a collection of modular and reusable compiler and
+toolchain technologies.
+
+This port adds links in /usr/bin to allow a devel/llvm* port to be
```
Comment (jrtc27) [inline devel/llvm-base/pkg-descr:4]: Should this be changed to "the devel/llvm port"?

File: devel/llvm-base/files/wrapper.sh.in (hunk 1 above)
Comment (brooksdavis) [reply devel/llvm-base/files/wrapper.sh.in:74]: So we don't need the `+a64c` and `+c64` any more? I've verified that we don't need the triples with existing packages.

File: devel/llvm-base/pkg-descr (hunk 3 above)
Comment (brooksdavis) [reply devel/llvm-base/pkg-descr:4]: will do

File: devel/llvm-base/files/wrapper.sh.in (hunk 1 above)
Comment (jrtc27) [reply devel/llvm-base/files/wrapper.sh.in:74]: +a64c was never needed (you add +noa64c if you _don't_ want capabilities), +c64 isn't as of the end of last week (when it finally got merged)

File: devel/llvm-base/files/wrapper.sh.in [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+{
+	if [ "$QUIET" -eq 0 ]; then
+		echo >&2 "$@"
+	fi
+}
+
+run()
+{
+	debug	# add space before normal multiline output
+	info "Running:" "$@"
```
Comment (jrtc27) [inline devel/llvm-base/files/wrapper.sh.in:33]: Maybe this should be debug not info, that or we default to QUIET? For ccc it makes sense to print the command, but as native cc it's noisy and odd.

**Review by jrtc27 — COMMENTED** [review]
Just the one additional comment upon rereading, I think

File: devel/llvm-base/files/wrapper.sh.in (hunk 4 above)
Comment (jrtc27) [reply devel/llvm-base/files/wrapper.sh.in:33]: Or, to be honest, perhaps we should have err and nothing else; debug is only used here by run (which is there to separate all the prior debug calls from the run line in ccc, none of which exist here), and info is only used by run, which should be at least off by default, if it's even worth having...

Comment (brooksdavis) [reply devel/llvm-base/files/wrapper.sh.in]: I've removed info (and QUIET) and switched to debug for the Running: line. A somewhat fancier version would look for -v and -### and set VERBOSE, but I'll leave that for later if it seems useful

File: devel/llvm-base/files/wrapper.sh.in [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+{
+	debug "Running:" "$@"
+	"$@"
+}
+
+
+tool=$0
+[ -L "$tool" ] && tool=$(/bin/realpath $tool)
+tool=${tool##*/}
+echo $tool
```
Comment (jrtc27) [inline devel/llvm-base/files/wrapper.sh.in:32]: _sigh_ missed this debugging line

**Review by jrtc27 — APPROVED** [review]
Didn't spot anything this time through so let's say it's good to go...

**Pull request merged**